### PR TITLE
Training data mode in Mutect2 to prepare for Mutect3

### PR DESCRIPTION
**Pull request by davidbenjamin**
@takutosato Could you look at this?

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/FeaturizedReadSets.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        final List<Integer> result = new ArrayList<>();
+        result.add(read.getMappingQuality());
+        result.add(BaseQuality.getBaseQuality(read, vc).orElse(DEFAULT_BASE_QUALITY));
+        result.add(read.isFirstOfPair() ? 1 : 0);
+        result.add(read.isReverseStrand() ? 1 : 0);
+
+        // distances from ends of read
+        final int readPosition = ReadPosition.getPosition(read, vc).orElse(0);
+        result.add(readPosition);
+        result.add(read.getLength() - readPosition);
```
Comment (takutosato) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/FeaturizedReadSets.java:122]: If this is the distance to either end of the read, `Math.min(readPosition, read.getLength() - readPosition)`

**Review by takutosato — APPROVED** [review]
Looks good

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/DepthPerSampleHC.java [hunk 2]
```diff
@@ -39,7 +39,7 @@
  *
  */
 @DocumentedFeature(groupName=HelpConstants.DOC_CAT_ANNOTATORS, groupSummary=HelpConstants.DOC_CAT_ANNOTATORS_SUMMARY, summary="Depth of informative coverage for each sample (DP)")
-public final class DepthPerSampleHC extends GenotypeAnnotation implements StandardHCAnnotation, StandardMutectAnnotation {
+public final class  DepthPerSampleHC extends GenotypeAnnotation implements StandardHCAnnotation, StandardMutectAnnotation {
```
[the remark this answers is not in the document]
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/DepthPerSampleHC.java]: true

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2IntegrationTest.java [hunk 3]
```diff
@@ -362,7 +387,7 @@ public void testForceCalling() {
             runMutect2(tumor, unfilteredVcf, "20:9998500-10010000", b37Reference, Optional.empty(),
                     args -> args.add(AssemblyBasedCallerArgumentCollection.FORCE_CALL_ALLELES_LONG_NAME, forceCalls),
                     args -> args.add(ReadThreadingAssemblerArgumentCollection.KMER_SIZE_LONG_NAME, kmerSize),
-                    args -> args.add(ReadThreadingAssemblerArgumentCollection.DONT_INCREASE_KMER_SIZE_LONG_NAME, true));
+                    args -> args.add(ReadThreadingAssemblerArgumentCollection.DONT_INCREASE_KMER_SIZE_LONG_NAME, true);
```
[the remark this answers is not in the document]
Comment (davidbenjamin) [reply src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2IntegrationTest.java]: true again

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/FeaturizedReadSets.java (hunk 1 above)
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/FeaturizedReadSets.java:122]: We're annotating both distances here — it's not the same as the `ReadPosition` annotation.

**Pull request merged**